library/config.c

```diff
@@ -56,31 +52,30 @@ int load_config_file(const char *filename, ziti_config **cfg) {
     }
     fclose(file);
 
-    ziti_config *c;
-    if (parse_ziti_config_ptr(&c, config, config_len) < 0) {
-        free_ziti_config(c);
+    if (parse_ziti_config(cfg, config, config_len) < 0) {
         free(config);
-        FREE(c);
         return ZITI_INVALID_CONFIG;
     }
 
-    *cfg = c;
     free(config);
-
     return ZITI_OK;
 }
 
-int load_config(const char *filename, ziti_config **cfg) {
-    if (filename != NULL) {
-        return load_config_file(filename, cfg);
+int load_config(const char *cfgstr, ziti_config *cfg) {
+    if (!cfgstr) {
+        return ZITI_INVALID_CONFIG;
+    }
+
+    memset(cfg, 0, sizeof(*cfg));
+    int rc = parse_ziti_config(cfg, cfgstr, strlen(cfgstr));
+
+    if (rc < 0) {
+        rc = load_config_file(cfgstr, cfg);
```

Comment:
maybe a DEBUG log here so we can tell that config parsing failed and are falling back to file loading?